src/Symfony/Component/Lock/Tests/Store/ZookeeperStoreTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Tests\Store;
+
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\Store\ZookeeperStore;
+use Exception;
+use Zookeeper;
+
+/**
+ * @author Ganesh Chandrasekaran <gchandrasekaran@wayfair.com>
+ *
+ * @requires extension zookeeper
+ */
+class ZookeeperStoreTest extends AbstractStoreTest
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function setupBeforeClass()
+    {
+        $zookeeper_server = getenv('ZOOKEEPER_HOST').':2181';
+
+        try {
+            $zookeeper = new Zookeeper(implode(',', array($zookeeper_server)));
+            // The session creation happens asynchronously, so we can wait for up to a second before giving up.
+            $attempt = 0;
+
+            while ($attempt < 10) {
+                if (Zookeeper::CONNECTED_STATE === $zookeeper->getState()) {
+                    return;
+                }
+                ++$attempt;
+                usleep(10000);
+            }
+
+            self::markTestSkipped('Unable to connect to the zookeeper host.');
+        } catch (Exception $ex) {
+            self::markTestSkipped('Unable to connect to the zookeeper host: '.$ex->getMessage());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @return ZookeeperStore
+     */
+    public function getStore(): ZookeeperStore
+    {
+        $zookeeper_server = getenv('ZOOKEEPER_HOST').':2181';
+
+        $zookeeper = new Zookeeper(implode(',', array($zookeeper_server)));
```

Comment:
How does Zookeeper handle integrity when using a cluster of servers?
Are you sure that creating an existing node would raise an exception even if 2 concurrent requests ask the creation at exactly the same time on 2 distinguished servers?

Reply:
Great question - Zookeeper always has only 1 leader and accepts writes through that leader and uses quorum, so even if it due to some race condition 2 process try to create a node, the second one will raise a warning and eventually`LockConflictedException` will be thrown.